File: map_ts/src/Entity/entity-contract.ts
export type ContractCondition = {
  name: string;
  satisfied: boolean;
};


export type OperationCost = {
  amount: number;
  unit: string;
};


export type EntityContract = {
  name: string;

  conditions: ContractCondition[];

  applyCost?: OperationCost;
  evolveCost?: OperationCost;
};

File: map_python/src/Entity/entity_contract.py
from dataclasses import dataclass, field


@dataclass(slots=True)
class ContractCondition:
    name: str
    satisfied: bool


@dataclass(slots=True)
class OperationCost:
    amount: float
    unit: str


@dataclass(slots=True)
class EntityContract:
    name: str
    conditions: list[ContractCondition] = field(default_factory=list)
    apply_cost: OperationCost | None = None
    evolve_cost: OperationCost | None = None
